module/platform/Webots/src/Webots.cpp

```diff
@@ -191,7 +191,7 @@ namespace module::platform {
 
         // Loop through the linked list of potential options for connecting. In order of best to worst.
         for (addrinfo* addr_ptr = address; addr_ptr != NULL; addr_ptr = addr_ptr->ai_next) {
-            int fd = socket(addr_ptr->ai_family, addr_ptr->ai_socktype, addr_ptr->ai_protocol);
+            fd = socket(addr_ptr->ai_family, addr_ptr->ai_socktype, addr_ptr->ai_protocol);
```

Comment:
This functions was not meant to directly modify the state of the reactor.
```suggestion
            int fd_temp = socket(addr_ptr->ai_family, addr_ptr->ai_socktype, addr_ptr->ai_protocol);
```

Reply:
I assume the other `fd`'s need to be changed to `fd_temp` too? All of them in the rest of this function?